_rules/links-with-identical-names-and-context-serve-equivalent-purpose-fd3a94.md

```diff
@@ -0,0 +1,298 @@
+---
+id: fd3a94
+name: Links with identical accessible names and context serve equivalent purpose
+rule_type: atomic
+description: |
+  This rule checks that links with identical accessible names and context resolve to the same resource or equivalent resources.
+accessibility_requirements:
+  wcag20:2.4.4: # Link Purpose (In Context) (A)
+    forConformance: true
+    failed: not satisfied
+    passed: further testing needed
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+authors:
+  - Anne Thyme Nørregaard
+---
+
+## Applicability
+
+This rule applies to any set of any two or more HTML or SVG elements that have the [semantic role](#semantic-role) of `link`, or a role that inherits from the `link` role, are [included in the accessibility tree](#included-in-the-accessibility-tree), and that have [matching](#matching-characters) [accessible names](#accessible-name) that do not only consist of [whitespace](#whitespace), and have [matching](#matching-characters) [programmatically determined link contexts](#programmatically-determined-link-context).
+
+**Note:** The test target for this rule is the full set of link elements that share the same [matching](#matching-characters) [accessible name](#accessible-name) and [matching](#matching-characters) [programmatically determined link contexts](#programmatically-determined-link-context).
+
+## Expectation
```

Comment:
According to SC 2.4.4 the purpose of each link can be determined from the link text alone or from the link text together with its programmatically determined link context.

I do not think the test rule tests this. According to the test rule, links with matching names should resolve to the same or equivalent resource. If the linked resource(s) have a different purpose than what is described in the link names, this would pass the test rule, but fail SC 2.4.4.

According to the SC 2.4.4 understanding article "it is a best practice for links with the same destination to have consistent text (and this is a requirement per Success Criterion 3.2.4 for pages in a set). It is also a best practice for links with different purposes and destinations to have different link text."

In my opinion, this test rule is more relevant for SC 3.2.4 Consistent Identification. Either way, SC 3.2.4 requires the identification to be consistent, but not necessarily identical.

Reply:
But this rule also allows for "equivalent resources" not just "same resource", so it goes beyond the best practice and is actually a good basis to fail the SC on if it doesn't link to the same or equivalent resource despite having the same accessible name and context right?